### ISSUE TITLE
Product Descriptions all Gone

### ISSUE THREAD

**Issue by miketurnermk** [facebookincubator/facebook-for-woocommerce#221]
Hi,

Suddenly all of my product descriptions have gone out of Facebook and been replaced with a basic description i.e. "English Bull Terrier - Green Fingers - Yes" (which is the combination of the product name and the variation)

This wasn't previously doing this and was showing the full description in the woocommerce product.

Has something just been changed?

The only way I seem to be able to force it to bring the right description now is either add it to the facebook description box in the product settings or to copy/paste the full product description in to each variation description box.

I didn't have to do before as it was just taking the original product description. I don't really relish the thought of having to manage it in that way, or going through all my products and copying and pasting a description multiple times + the extra overheads of then having to go back through them if I need to tweak it.

...help!

https://www.facebook.com/pg/OddBulls/shop/

**Comment by tahoff** [comment]
Hmm, did you update anything such as the Facebook-specific description? On our end it appears the product was last updated on 1/14, are you sure that this product didn't always have this description on your Facebook shop?

**Comment by miketurnermk** [comment]
No, definitely didn't touch the Facebook description (other than for a few items last night to test this behaviour again)

I'm 95% certain that when I first setup the store I tested all of the products and was seeing a long description in each, but now I just see a few words (the title suffixed by the variation)

I raised other issues in the last few days around the topic of no unique descriptions being shown

It was only after setting up another store feed last night that I found the descriptions weren't going in. At first I thought it was something unique to the site but then went back to my original site and found the same.

What is the designed behaviour for a product which has variations and does not have another description populated in the variation description or facebook description? - shouldn't it be doing what I am expecting? - i.e. to just take the overall product description and feed that through?

**Comment by tahoff** [comment]
Do you use a specific extension for the personalization aspect of your products?

**Comment by miketurnermk** [comment]
I use WC Fields Factory to add custom fields into the products. However, this doesn't touch the description at all (it just adds extra fields in).  Also, products that don't feature any personalisation / call to the plugin are also affected.

**Comment by tahoff** [comment]
@miketurnermk, which products without personalization are affected? I see the following products without personalization: 
https://www.facebook.com/commerce/products/1579754108767494/
https://www.facebook.com/commerce/products/1658787034164746/
https://www.facebook.com/commerce/products/2315037658522129/
https://www.facebook.com/commerce/products/1791117537568027/
which don't have the abbreviated description.

And even if they don't visibly touch the description, it can generate incompatibilities.

**Comment by miketurnermk** [comment]
For example, in my other store:

https://www.facebook.com/commerce/products/1998445910172263/
https://www.facebook.com/commerce/products/1209647649138341/

Both of these have no personalisation feature and have the incorrect description.

How can we prove the issue? - I'm assuming there is some logic which says IF field = x then populate fb description with y ELSE add title + variation in FB description?

Is there a field that I can check in the DB against the impacted products?

Also, is there a URL on my woocommerce store where I can actually see the output of the FB feed itself? (rather than having to resort to FB to see it)

**Comment by tahoff** [comment]
@miketurnermk, thanks for those links.

Unfortunately, we don't actually create a feed (this integration uses the API), there isn't a feed file to look at. So looking at your product catalog or the products in your shop are the only ways to see the product info. 

We don't do any parsing like that and we haven't seen this issue in our demo stores, which is why this is troubling. 

Can you email ads_extension_woocommerce@fb.com with screenshots of the Woo product page of an affected product (variable and non-variable)? Additionally, a screenshot of the extensions you have installed would be tremendously helpful.

Any differences that you know about between the products with the shorter descriptions and the longer descriptions would be tremendously helpful. Finally, if that doesn't work, maybe try Deleting settings in the Facebook Ads Extension pop-up window to clear the old product information and then re-syncing.

**Comment by miketurnermk** [comment]
Thanks @tahoff  - have just sent an email through

**Comment by evenmed** [comment]
I'm having a similar issue. In my case, it's the variable products that don't get a description, just a combination of the product name and selected variation. Simple products work fine, but every single variable product has the error.

Has a solution for this been found?

**Comment by miketurnermk** [comment]
I haven't heard anything back since I was asked to send an email with more details 12 days ago.  Since then I just had to manually go through all the products and copy/paste the description to the Facebook product field which was a pain (anytime I tweak the description I also have to remember to tweak the fb description too).

**Comment by evenmed** [comment]
Yeah that's what I was thinking of doing as a temporary solution as well, hopefully gets fixed soon.

I guess what would need to be done would be to check if the product is a variation when the description is being fetched, and if it is, get the parent product's description instead. I'll dive into the code and submit a pull request if I find a solution.

**Comment by StefanVB** [comment]
I am having a similar issue. I would place plain text product description in the "Facebook" tab within the "Facebook Description" box but on saving the text block would go blank again. The "simple product" type work however the "simple subscription" ([Woocommerce Subscriptions Extension](https://woocommerce.com/products/woocommerce-subscriptions/)) does not. 

This has resulted in Facebook product descriptions which display short code text instead of plain text descriptions (see [link](https://www.facebook.com/commerce/products/1735975699779092/) )

**Comment by evenmed** [comment]
@StefanVB I think you're experiencing a different issue. If I understand correctly, what's happening to @miketurnermk and myself is that the descriptions of variable products don't get imported into the Facebook product automatically. However, manually setting the description in Product Data > Facebook > Facebook Description works fine.

If your issue is that manually setting the description in Product Data > Facebook > Facebook Description doesn't work, you should probably submit a new issue.

**Comment by evenmed** [comment]
After coding a solution, I just found out that this was already fixed 3 hours ago with version 1.7.10... https://github.com/facebookincubator/facebook-for-woocommerce/commit/1d5792ea39deb3ab23c196754d0e876e7f30b651

I just updated the plugin and resynced the products and all the variable product's descriptions were imported correctly now.

Glad this is solved.

**Comment by mengyingdu** [comment]
This was fixed in v1.7.10. Can you upgrade latest v1.7.11 to confirm whether your problem still exist?

**Comment by dmitridr** [comment]
Marking this as solved. Please open a new issue and reference this one if the problem persists.
